python-spec/src/somacore/data.py

```diff
@@ -46,15 +46,26 @@ def create(
         platform_config: Optional[options.PlatformConfig] = None,
         context: Optional[Any] = None,
     ) -> Self:
-        """Creates a new DataFrame."""
+        """Creates a new ``DataFrame`` at the given URI.
+
+        :param uri: The URI where the ``DataFrame`` will be created.
+        :param schema: Arrow schema defining the per-column schema. This schema
+            must define all columns, including columns to be named as index
+            columns.  If the schema includes types unsupported by the SOMA
+            implementation, an error will be raised.
+        :param index_column_names: A list of column names to use as user-defined
+            index columns (e.g., ``['cell_type', 'tissue_type']``).
+            All named columns must exist in the schema, and at least one
+            index column name is required.
+        """
         raise NotImplementedError()
 
     # Data operations
 
     @abc.abstractmethod
     def read(
         self,
-        coords: Optional[options.SparseDFCoords] = None,
+        coords: Optional[options.SparseDFCoords] = (),
```

Comment:
Since we now have the default of `()` that unambiguously means "everything", does this need to be `Optional` (i.e., no longer accept `None` as a value)? Consider: `DenseNDCoords` is currently *not* marked as `Optional`.

Reply:
as long as it's optional put-nothing-at-all (which it is) that's is fine with me

Reply:
Correct:

✅ `some_df.read()`
✅ `some_df.read(coords=())`
❌ `some_df.read(coords=None)`

I’ve put this change in for now as its own commit so if we don’t want it, it’s easy to back out.

Reply:
@bkmartinjr (and any other readers): another thing to check in on (goes with https://github.com/single-cell-data/SOMA/pull/119#discussion_r1103107462)

Reply:
I am _very_ good with this change.